### PR TITLE
migrate java8: remove unnecessary cast expressions

### PR DESCRIPTION
**Pull request by thinkerou**
Motivation:

remove unnecessary cast expressions

### CODE REVIEW

**Comment by normanmaurer** [comment]
@thinkerou seems like you introduced a bug during refactoring. Please fix and let  me know once done

**Comment by thinkerou** [comment]
Error info:
```
Stacktrace
java.lang.StackOverflowError
	at io.netty.handler.codec.http.DefaultHttpHeaders.contains(DefaultHttpHeaders.java:327)
```

Source:
```
    @Override
    public boolean contains(String name, String value, boolean ignoreCase) {
        return contains((CharSequence) name, (CharSequence) value, ignoreCase);
    }

    @Override
    public boolean contains(CharSequence name, CharSequence value, boolean ignoreCase) {
        return headers.contains(name, value, ignoreCase ? CASE_INSENSITIVE_HASHER : CASE_SENSITIVE_HASHER);
    }
```
If change `return contains((CharSequence) name, (CharSequence) value, ignoreCase);` to `return contains(name, value, ignoreCase);`, it will recursive call self.
so I revert the line.

**Comment by normanmaurer** [comment]
@netty-bot test this please

File: codec/src/test/java/io/netty/handler/codec/bytes/ByteArrayDecoderTest.java [hunk 1]
```diff
@@ -40,13 +40,13 @@ public void testDecode() {
         byte[] b = new byte[2048];
         new Random().nextBytes(b);
         ch.writeInbound(wrappedBuffer(b));
-        assertThat((byte[]) ch.readInbound(), is(b));
+        assertThat(ch.readInbound(), is(b));
```
[the remark this answers is not in the document]
Comment (normanmaurer) [reply codec/src/test/java/io/netty/handler/codec/bytes/ByteArrayDecoderTest.java]: @thinkerou please check and also please rebase.

[the remark this answers is not in the document]
Comment (thinkerou) [reply codec/src/test/java/io/netty/handler/codec/bytes/ByteArrayDecoderTest.java]: @johnou @normanmaurer revert it done.

**Comment by johnou** [comment]
To be honest I'm not sure what this pull request buys us, we are removing the casts which now require you to look at the object / getter to see what type it is.. what's the upside?

**Comment by thinkerou** [comment]
@johnou you are right, I will close the pr. thanks! not much benefit.

**Comment by doom369** [comment]
> the casts which now require you to look at the object / getter to see what type it is

In some cases you have the variable type, so you always know the variable type, here for example:

`FullHttpResponse response = (FullHttpResponse) ch.readOutbound();`

so for such cases this is good PR.

**Comment by johnou** [comment]
@doom369 yes that's an example I can tolerate :+1: on the other hand something like this..

```java
                http2Server.encoder().writeHeaders(ctx,
                        - (Integer) invocationOnMock.getArgument(1),
                        - (Http2Headers) invocationOnMock.getArgument(2),
                        + invocationOnMock.getArgument(1),
                        + invocationOnMock.getArgument(2),
                        0,
                        false,
                        ctx.newPromise());
```

would be a bit painful debugging as the arg types are not immediately visible.

**Comment by thinkerou** [comment]
closing.

**Pull request closed without merging**